src/py21cmmc/likelihood.py

```diff
@@ -1782,3 +1882,136 @@ def computeLikelihood(self, model):
+        if k is not None:
+            self.k = k
+            self.k_len = len(k)
+        if datafile is not None:
+            self.datafile = [datafile]
+            self.data = dict(np.load(datafile))
+            if k is None:
+                self.k = [self.data['kwfband10'], self.data['kwfband8']]
+                self.k_len = max(len(i) for i in self.k)
+            
+
+    @classmethod
+    def from_builtin_data(cls, datafile = "", **kwargs):
+        datafile = path.join(path.dirname(__file__), "data", datafile + ".npz")
+        
+        return cls(datafile = datafile, **kwargs)
+        
+    def reduce_data(self, ctx):
+        # Interpolate the data onto the HERA bands and ks
+        final_PS = np.zeros((len(self.redshifts), self.k_len))
+        for i in range(self.redshifts.shape[0]):
+            interp_ks = self.k[i]
+            final_PS[i,:len(interp_ks)] = interp2d(ctx.get("k"),ctx.get("ps_redshifts"), ctx.get('delta'))(interp_ks, self.redshifts[i])
+        final_data = {'k' : self.k,
+                      'delta' : final_PS,
+                     }
+        # If we are using the emualtor, include the error.
+        try:
+            final_PS_err = np.zeros((len(self.redshifts), self.k_len))
+            for i in range(self.redshifts.shape[0]):
+                interp_ks = self.k[i]
+                final_PS_err[i,:len(interp_ks)] = interp2d(ctx.get("k"),ctx.get("ps_redshifts"), ctx.get("delta_err"))(interp_ks, self.redshifts[i])
+            final_data['delta_err'] = final_PS_err
+        except:
+            pass
+        return [final_data]
+
+    def computeLikelihood(self, model):
+        """
+        Compute the likelihood given 1D power spectrum values at the same k-bins as the data.
+
+        Parameters
+        ----------
+        model : np.ndarray
+            1D Power spectrum in log10(mK^2) and its k bin values in /Mpc
+            If there is a window function available, then input the PS after the WF has been applied.
+
+        Returns
+        -------
+        lnl : float
+            Log likelihood for the provided model.
+        Data shape = 5 fields, 37 kbins, 4 = [kval, power, lower, upper limit error], 2 (band1=10 band2=8)
+        """
+
+        lnl = 0
+        for band in self.data[0]['z_bands']:
+            for field in range(self.data[0]['band8'].shape[0]):
+                PS_limit_ks = self.data[0]['band' + str(round(band))][field,:,0]
+                PS_limit_ks = PS_limit_ks[~np.isnan(PS_limit_ks)]
+                Nkbins = len(PS_limit_ks)
+                PS_limit_vals = self.data[0]['band' + str(round(band))][field,:Nkbins,1] 
+                PS_limit_vars = self.data[0]['band' + str(round(band))][field,:Nkbins,2]  
+
+                kwf_limit_vals = self.data[0]['kwfband' + str(round(band))]
+                Nkwfbins = len(kwf_limit_vals)
+                PS_limit_wfcs = self.data[0]['wfband' + str(round(band))][field,:Nkbins,:]
+                mask = ~np.isnan(PS_limit_wfcs)
```

Comment:
does the mask really do anything here? If so, you won't be able to reshape below